lib/workout.js

```diff
@@ -30,12 +30,29 @@ module.exports = class Workout extends Content {
     this.parent = null;
   }
 
+/**
+ * Take text of a Workout and return an object representing it
+ * @param {string} Text of Workout file
+ */
   parse(text) {
-    yaml.safeLoadAll(text.split('---')[0], (doc) => {
-      for (const prop in doc) {
-        this[prop] = doc[prop];
+    var exerciseArr = text.split('---');
+    var data = yaml.safeLoad(exerciseArr.shift());
+
+    data.exercises = exerciseArr.map((exercise)=> {
+      // Destructuringly assigning data into metadata and question
+      var [exerciseMetadata, question] = exercise.split("### Question\n").map(x => {return x.trim()});
+      exerciseMetadata = yaml.safeLoad(exerciseMetadata);
+
+      // Accept ___-link key and pull `link-type` and `link` props from it
+      for (let key in exerciseMetadata) if (key.indexOf('-link') != -1 ) {
+        exerciseMetadata["link-type"] = key.split('-').shift();
+        exerciseMetadata["link"] = exerciseMetadata[key];
+        delete exerciseMetadata[key];
       }
-    });
+
+      return {...exerciseMetadata, question};
+    })
+    return data;
```

Comment:
This returns data, but does not populate`this`, which means you're changing the interface. Append yaml-derived metadata and exercises to `this`. You can still return the parsed data, but `parse` should populate the class.